src/quacc/recipes/newtonnet/ts.py

```diff
@@ -293,3 +307,250 @@ def _get_hessian(atoms: Atoms) -> NDArray:
     ml_calculator.calculate(atoms)
 
     return ml_calculator.results["hessian"].reshape((-1, 3 * len(atoms)))
+
+
+"""
+@job
+@requires(NewtonNet, "NewtonNet must be installed. Refer to the quacc documentation.")
+def neb_job(
+    atoms: Atoms,
+    relax_endpoints: bool = True,
+    run_geodesic: bool = True,
+    run_single_ended: bool = True,
+    run_freq: bool = True,
+    neb_job_kwargs: dict[str, Any] | None = None,
+    relax_job_kwargs: dict[str, Any] | None = None,
+    freq_job_kwargs: dict[str, Any] | None = None,
+):
+    relax_job_kwargs = relax_job_kwargs or {}
+    freq_job_kwargs = freq_job_kwargs or {}
+
+    irc_job_defaults = {"max_steps": 5}
+    irc_job_kwargs = recursive_dict_merge(irc_job_defaults, irc_job_kwargs)
+
+    # Run IRC
+    irc_summary = strip_decorator(irc_job)(
+        atoms, direction=direction, run_freq=False, **irc_job_kwargs
+    )
+
+    # Run opt
+    relax_summary = strip_decorator(relax_job)(irc_summary["atoms"], **relax_job_kwargs)
+
+    # Run frequency
+    freq_summary = (
+        strip_decorator(freq_job)(relax_summary["atoms"], **freq_job_kwargs)
+        if run_freq
+        else None
+    )
+    relax_summary["freq_job"] = freq_summary
+    relax_summary["irc_job"] = irc_summary
+
+    return relax_summary
+"""
+
+
+def sella_wrapper(
+    atoms_object,
+    traj_file=None,
+    sella_order=0,
+    use_internal=True,
+    traj_log_interval=2,
+    fmax_cutoff=1e-3,
+    max_steps=1000,
+):
+    if traj_file:
+        traj = Trajectory(traj_file, "w", atoms_object)
+    qn = Sella(atoms_object, order=sella_order, internal=use_internal)
+    if traj_file:
+        qn.attach(traj.write, interval=traj_log_interval)
+    qn.run(fmax=fmax_cutoff, steps=max_steps)
+    if traj_file:
+        traj.close()
+
+
+def geodesic_interpolate_wrapper(
+    r_p_atoms: Atoms,
+    nimages: int = 17,
+    sweep: bool | None = None,
```

Comment:
Seems like this should just be `bool = False`

Reply:
Hi @Andrew-S-Rosen ,
This seems like an interesting use of default value structures. It decides whether to call sweep or smooth depending on the size of the system but only when the value provided is None. Otherwise choose whatever `bool` is provided as the argument.
Please let me know if it makes sense. I have followed a structure that was used in the geodesic_interpolate's repository but they just didn't have a wrapper function like this.
Thanks.

Reply:
In that case, it would likely be clearer to have something like `sweep: bool | Literal["auto"] = "auto"`.